pages/reset-password/index.tsx

```diff
@@ -0,0 +1,235 @@
+import Image from "next/image";
+import React, { useEffect, useState } from "react";
+import { useRouter, NextRouter } from "next/router";
+import Link from "next/link";
+import Input from "@/components/primitives/Form/Input";
+import Button from "@/components/primitives/Button/button";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { z, ZodError, ZodType } from "zod";
+
+// Define custom API error type
+type ApiErrorResponse = {
+  status: "Error";
+  message: string;
+};
+
+interface FormValues {
+  password: string;
+  confirmPassword: string;
+}
+
+type formData = {
+  password: string;
+  confirmPassword: string;
+};
+
+interface QueryParams {
+  token?: string;
+}
+
+// Define Zod schema for password and confirmPassword validation
+const passwordSchema: ZodType<formData> = z
+  .object({
+    password: z
+      .string()
+      .min(8, {
+        message: "Password must must not be less than 8 characters",
+      })
+      .refine((value) => /[!@#$%^&*(),.?":{}|<>]/.test(value), {
+        message: "Password must contain at least one special character",
+      })
+      .refine((value) => /\d/.test(value), {
+        message: "Password must contain at least one digit",
+      })
+      .refine((value) => /[a-z]/.test(value), {
+        message: "Password must contain at least one lowercase letter",
+      })
+      .refine((value) => /[A-Z]/.test(value), {
+        message: "Password must contain at least one uppercase letter",
+      }),
+    confirmPassword: z.string(),
+  })
+  .refine((data) => data.password === data.confirmPassword, {
+    message: "Passwords do not match",
+    path: ["confirmPassword"],
+  });
+
+const ResetPassword: React.FC = () => {
+  const router: NextRouter = useRouter();
+  const [serverError, setServerError] = useState("");
+  const { token }: QueryParams = router.query as QueryParams;
+
+  const {
+    handleSubmit,
+    register,
+    formState: { errors: formErrors },
+    setError,
+    clearErrors,
+  } = useForm<FormValues>({});
+
+  const onSubmit: SubmitHandler<FormValues> = async (data) => {
+    try {
+      // // Validate using Zod schema
+      await passwordSchema.parseAsync({
+        confirmPassword: data.confirmPassword,
+        password: data.password,
+      });
+
+      const response = await fetch(
+        "https://abeghelp-backend-staging.up.railway.app/api/v1/auth/password/reset",
+        {
+          method: "POST",
+          headers: {
+            "content-type": "application/json",
+          },
+          body: JSON.stringify({
+            token,
+            password: data.password,
+            confirmPassword: data.confirmPassword,
+          }),
+        },
+      );
+
+      if (response.ok) {
+        // Redirect or handle success
+        await new Promise<void>((resolve) => setTimeout(resolve, 2000));
+        await router.push("/reset-password/confirmation");
+      } else {
```

Comment:
If response is not ok, wouldn't that break out of the try and go to the catch? 

If yes, that means this else statement is redundant because the catch block has an error handling process again